README.md

```diff
@@ -5271,69 +5335,111 @@ sleep 30
+
+- `critic.exp_manager.checkpoint_callback_params.every_n_train_steps` should be set to `actor.trainer.val_check_interval * actor.model.global_batch_size / critic.model.global_batch_size` so that the Critic is saved at the same frequency as the Actor.
+- `critic.inference.micro_batch_size` should be set to `actor.model.rlhf.ppo.rollout_micro_batch_size` divided by the Critic's data parallel size (which is obtained by the total number of GPUs the Critic is running on, i.e., `trainer.devices * trainer.num_nodes`, divided by the product of `model.tensor_model_parallel_size * model.pipeline_model_parallel_size`), rounded up.
+This ensures that the Critic can process the Actor's requests as efficiently as possible.
+- Similarly, `rm.inference_micro_batch_size` and `init_policy.inference_micro_batch_size` should be set to `actor.model.rlhf.ppo.rollout_micro_batch_size` divided by the RM and Initial Policy's data parallel size, rounded up.
+- `critic.model.ppo_epochs` should be equal to `actor.model.rlhf.ppo.epochs` so that the Critic performs the same number of updates as the Actor on the rollout buffer data.
+
+##### 5.16.2.7 PPO Hyper-parameters
 <a id="markdown-ppo-hyper-parameters" name="ppo-hyper-parameters"></a>
 
-All the model related parameters can be controlled the same way as in other NeMo training jobs. However, we also provide full control of the behavior of PPO during training, with a section in the config yaml files inside `model.rlhf`. These are the descriptions of the available hyper-parameters:
+All the model parameters can be controlled the same way as in other NeMo training jobs. However, we also provide full control of the behavior of PPO during training, with a section in the Actor config yaml file inside `model.rlhf`. These are the available hyper-parameters:
 
-- `rlhf.reward_model`: Provide the ip address and the port where the Reward Model will be running, to enable communication with it.
-- `rlhf.critic`: Provide the ip address and the port where the PPO Critic will be running, to enable communication with it.
-- `rlhf.initial_policy`: Provide the ip address and the port where the Initial Policy will be running, to enable communication with it.
-- `rlhf.ppo.entropy_penalty`: Control the effect of the entropy term in PPO.
-- `rlhf.ppo.inital_pollicy_kl_penalty`: Control the effect of the initial policy KL Divergence term in PPO.
-- `rlhf.ppo.use_absolute_kl`: Whether to use the absolute value of the initial policy KL Divergence or not.
-- `rlhf.ppo.epochs`: Number of epochs the actor and critic will perform on the data stored in the rollout buffer each time.
+- `rlhf.{reward_model,critic,initial_policy}`: Provide the ip address and the port where the Reward Model, PPO Critic and Initial Policy will be running, to enable communication with them.
```

Comment:
how about rlhf.{reward_model,critic,initial_policy}.{ip,port} here?

Reply:
Yes good idea, done in 3834ce9d